src/x/debugdump/host.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package debugdump
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+)
+
+// hostDataProvider is an InfoProvider implementation returning data about the underlying host:
+// PID, working directory, etc.
+type hostDataProvider struct{}
+
+type hostInformation struct {
+	PID int    `json:"pid"`
+	CWD string `json:"cwd"`
+}
+
+// NewHostDataProvider returns a DataProvider for host information
+func NewHostDataProvider() DataProvider {
+	return &hostDataProvider{}
+}
+
+// ProvideData provides data about the host and writes it in the given writer
+func (h *hostDataProvider) ProvideData(w io.Writer) error {
+	wd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	hostInfo := &hostInformation{
+		PID: os.Getpid(),
+		CWD: wd,
+	}
+	jsonData, err := json.Marshal(hostInfo)
```

Comment:
Wanna use [MarshalIndent](https://stackoverflow.com/questions/19038598/how-can-i-pretty-print-json-using-go) to get slightly easier to read output?